databricks_cli/stack/api.py

```diff
@@ -239,8 +254,62 @@ def _update_job(self, job_settings, job_id):
+        :param resource_properties: dict of properties for the workspace asset. Must contain the
+        'source_path' and 'path' fields. The other fields will be inferred if not provided.
+        :param physical_id: dict containing physical identifier of workspace asset on databricks.
+        Should contain the field 'path'.
+        :param overwrite: Whether or not to overwrite the contents of workspace notebooks.
+        :return: (dict, dict) of (physical_id, deploy_output). physical_id is the physical ID for
+        the stack status that contains the workspace path of the notebook or directory on datbricks.
+        deploy_output is the initial information about the asset on databricks at deploy time
+        returned by the REST API.
+        """
+        # Required fields. TODO(alinxie) put in _validate_config
+        local_path = resource_properties.get('source_path')
+        workspace_path = resource_properties.get('path')
+        object_type = resource_properties.get('object_type')
+
+        actual_object_type = 'DIRECTORY' if os.path.isdir(local_path) else 'NOTEBOOK'
+        if object_type != actual_object_type:
+            raise StackError("Field 'object_type' ({}) not consistent"
+                             "with actual object type ({})".format(object_type, actual_object_type))
+
+        click.echo('sync {} {} to {}'.format(object_type, local_path, workspace_path))
+        if object_type == 'NOTEBOOK':
+            # Inference of notebook language and format
+            lang_fmt = WorkspaceLanguage.to_language_and_format(local_path)
+            if lang_fmt is None:
+                raise StackError("Workspace Notebook language and format cannot be inferred"
+                                 "Please check file extension of notebook file.")
+            language, fmt = lang_fmt
+            # Deployment
+            self.workspace_client.mkdirs(
+                os.path.dirname(workspace_path))  # Make directory in workspace if not exist
+            self.workspace_client.import_workspace(local_path, workspace_path, language, fmt,
+                                                   overwrite)
+        elif object_type == 'DIRECTORY':
+            self.workspace_client.import_workspace_dir(local_path, workspace_path, overwrite,
+                                                       exclude_hidden_files=True)
+        else:
+            raise StackError("Invalid value for 'object_type' field: {}".format(object_type))
+
+        if physical_id and physical_id['path'] != workspace_path:
+            # physical_id['path'] is the workspace path from the last deployment. Alert when changed
+            click.echo("Workspace asset had path changed from {} to {}".format(physical_id['path'],
+                                                                               workspace_path))
+        new_physical_id = {'path': workspace_path}
```

Comment:
put spaces around the curly braces `{ 'path': workspace_path }`

Reply:
I get a pep8 violation when I do that...

Reply:
Ah python...